### PR TITLE
Changes to learned to optionally show sets of skills instead of all skills.

### PR DESCRIPTION
**Pull request by jayRyan24**
```
>;learned weapons
--- Lich: learned active.
   Large Blunt      0.31  (0.07/hr, 1.77/day)   Small Blunt      0.31  (0.07/hr, 1.77/day)
      Staves        0.31  (0.07/hr, 1.77/day)     Polearms       0.31  (0.07/hr, 1.77/day)
   Small Edged      0.3   (0.07/hr, 1.71/day) Twohanded Blunt    0.3   (0.07/hr, 1.71/day)
   Light Thrown     0.3   (0.07/hr, 1.71/day) Twohanded Edged    0.29  (0.07/hr, 1.65/day)
     Crossbow       0.28  (0.07/hr, 1.59/day)      Slings        0.28  (0.07/hr, 1.59/day)
       Bow          0.27  (0.06/hr, 1.54/day)     Brawling       0.27  (0.06/hr, 1.54/day)
  Offhand Weapon    0.26  (0.06/hr, 1.48/day)   Heavy Thrown     0.26  (0.06/hr, 1.48/day)
 Missile Mastery    0.25  (0.06/hr, 1.42/day)  Parry Ability     0.23  (0.05/hr, 1.31/day)
  Melee Mastery     0.22  (0.05/hr, 1.25/day)
   TOTAL RANKS: 4.75  HOURLY: 1.13  DAILY: 27.01
   TOTAL TDPS: 27.41  DAILY: 156.07
   Learning For: 4.2 hours
--- Lich: learned has exited.
```

### CODE REVIEW

File: learned.lic [hunk 1]
```diff
@@ -4,27 +4,105 @@
...
+  end
 
-DRSkill.list.select { |item| item.current - item.baseline > 0 }.sort_by { |item| item.current - item.baseline }.reverse.each_slice(columns) do |skills|
-  output = skills.map { |skill| format_skill_data(skill, learning_time) }.join
-  respond(output)
-end
+  def setup
+    arg_definitions = [
+      [
+        { name: 'combats', regex: /(combat|comb)/, optional: true, description: 'Show exp gained in combat skills.' },
```
Comment (rpherbig) [inline learned.lic]: Can you call this with multiple args? E.g. `;learned combats survival`? If I'm reading the `if`/`elsif` block starting on line 38, it would only report one type at a time.

File: learned.lic [hunk 2]
```diff
@@ -4,27 +4,105 @@
...
+      @skills_to_show.concat(survival_skills)
+                     .concat(lore_skills)
+                     .concat(armor_skills)
+                     .concat(weapon_skills)
+                     .concat(magic_skills)
+                     .concat(guild_skills)
+    end
+  end
+
+  pause
```
Comment (rpherbig) [inline learned.lic]: Why is this `pause` here?

File: learned.lic [hunk 3]
```diff
@@ -4,27 +4,105 @@
...
+  def display_data
+    learning_time = ((Time.now - DRSkill.start_time) / 60.0 / 60)
+    columns = get_settings.learned_column_count
 
-daily_tdps = tdps / learning_time * 24
+    DRSkill.list.select { |item| @skills_to_show.include?(item.name)}
+                .select { |item| item.current - item.baseline > 0 }
+                .sort_by { |item| item.current - item.baseline }
+                .reverse.each_slice(columns) do |skills|
+                  output = skills.map { |skill| format_skill_data(skill, learning_time) }.join
```
Comment (rpherbig) [inline learned.lic]: Note to self: I think this `.map` can be pulled out of the `.each_slice`. Outside the scope of this PR, I'll test it after we merge this.

File: learned.lic (hunk 1 above)
Comment (jayRyan24) [reply learned.lic]: It only shows one type at a time, you'd have to rerun it.

File: learned.lic (hunk 2 above)
Comment (jayRyan24) [reply learned.lic]: I am not sure, it was there initially so I left it in.

File: learned.lic (hunk 1 above)
Comment (jayRyan24) [reply learned.lic]: Are you asking me to implement this or asking if it does already support this?

Comment (rpherbig) [reply learned.lic]: Sorry for the confusion, let me rephrase.

The `arg_definitions` say that you can call it with any number of those optional args (e.g. `;learned combats survival`). But the `if`/`elsif` block prevents that from actually working.

What do you think about changing the `arg_definitions` to this?
```
    arg_definitions = [
      [ { name: 'combats', regex: /(combat|comb)/, description: 'Show exp gained in combat skills.' } ],
      [ { name: 'survival', regex: /(survival|surv)/, description: 'Show exp gained in survival skills.' } ],
      [ { name: 'magic', regex: /magic/, description: 'Show exp gained in magic skills.' } ],
      [ { name: 'armor', regex: /armor/, description: 'Show exp gained in armor skills.' } ],
      [ { name: 'weapons', regex: /(weapons|weap)/, description: 'Show exp gained in weapon skills.' } ],
      [ { name: 'lore', regex: /lore/, description: 'Show exp gained in lore skills.' } ],
      [ { name: 'guild', regex: /guild/, description: 'Show exp gained in your guild skill.'} ],
      []
    ]
```

Comment (jayRyan24) [reply learned.lic]: Then it requires at least one?

Comment (rpherbig) [reply learned.lic]: The final `[]` _should_ allow it to run without any args. I can't test that at the moment.
